main.go

```diff
@@ -79,12 +106,16 @@ func main() {
 		errChan <- fmt.Errorf("%s", <-sig)
 	}()
 
-	objList := newObjectList()
-	ready := make(chan struct{})
+	jobs := make(chan []int)
+	seen := make(map[int]struct{})
+	result := make(chan ObjectDetail)
```

Comment:
Please do not use unbuffered chans where you need to make some work and store result. Because it will be blocking